core/src/events/dispatcher.rs

```diff
@@ -43,8 +42,9 @@ impl Dispatcher {
     match event {
       WindowEvent::ModifiersChanged(s) => self.info.modifiers = s,
       WindowEvent::CursorMoved { position, .. } => {
-        let pos = position.to_logical::<f32>(wnd_factor);
-        self.cursor_move_to(Point::new(pos.x, pos.y), tree)
+        let logical_pos =
+          ScaleToLogic::new(1.0 / wnd_factor as f32).transform_point(position.cast());
```

Comment:
the core window does not need `ScaleToLogic` and `window_factor`,  `wnd_factor` should transparent to the core window. The s
hell window converts all positions to the logic axis before passing it to the core window.

Reply:
Does this mean the distinction between physical and logical should not exist at all within Ribir, but always be resolved in `ribir_winit`? Or did you mean just this place?
The distinction comes mostly from the part of `ribir_painter` that I moved out to `ribir_geometry`.

Reply:
Logic means scale, not axis, right?
ribir_painter used `DeviceSize` (physic)
I'm not sure how this can work. When everything is converted to logic before passed to the core window, does this need to converted to physics before passed to painter? That doesn't seem right to me.
Logic values use float. I'm not sure how that works with painting.
Can you please shed some light on that?